src/main/java/com/duckblade/osrs/toa/features/tomb/RewardSarcophagus.java

```diff
@@ -0,0 +1,277 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.duckblade.osrs.toa.features.tomb;
+
+import com.duckblade.osrs.toa.TombsOfAmascutConfig;
+import com.duckblade.osrs.toa.module.PluginLifecycleComponent;
+import com.duckblade.osrs.toa.util.RaidRoom;
+import com.duckblade.osrs.toa.util.RaidState;
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.Client;
+import net.runelite.api.Model;
+import net.runelite.api.WallObject;
+import net.runelite.api.events.WallObjectDespawned;
+import net.runelite.api.events.WallObjectSpawned;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.EventBus;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.events.ConfigChanged;
+
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+public class RewardSarcophagus implements PluginLifecycleComponent
+{
+
+	private static final int[] VARBIT_MULTILOC_IDS_CHEST = new int[]{
+		14356, 14357, 14358, 14359, 14360, 14370, 14371, 14372
+	};
+
+	private static final int VARBIT_VALUE_CHEST_KEY = 2;
+	private static final int VARBIT_ID_SARCOPHAGUS = 14373;
+	private static final int WALL_OBJECT_ID_SARCOPHAGUS = 46221;
+
+	private final EventBus eventBus;
+	private final Client client;
+	private final TombsOfAmascutConfig config;
+	private final ClientThread clientThread;
+
+	private final List<WallObject> wallObjects = new ArrayList<>();
+
+	@Nullable
+	private int[] defaultFaceColors1;
+	@Nullable
+	private int[] defaultFaceColors2;
+	@Nullable
+	private int[] defaultFaceColors3;
+
+	private boolean sarcophagusIsPurple;
+	private boolean purpleIsMine = true;
+
+	@Override
+	public boolean isEnabled(final TombsOfAmascutConfig config, final RaidState raidState)
+	{
+		return raidState.getCurrentRoom() == RaidRoom.TOMB;
+	}
+
+	@Override
+	public void startUp()
+	{
+		eventBus.register(this);
+		parseVarbits();
+	}
+
+	@Override
+	public void shutDown()
+	{
+		eventBus.unregister(this);
+		resetFaceColors(wallObjects);
+		wallObjects.clear();
+		defaultFaceColors1 = null;
+		defaultFaceColors2 = null;
+		defaultFaceColors3 = null;
+		sarcophagusIsPurple = false;
+		purpleIsMine = true;
+	}
+
+	@Subscribe
+	public void onConfigChanged(final ConfigChanged event)
+	{
+		if (!event.getGroup().equals(TombsOfAmascutConfig.CONFIG_GROUP) || wallObjects.isEmpty()) return;
+
+		final String key = event.getKey();
+
+		if (key.startsWith("sarcophagus"))
```

Comment:
I'd rather have key name constants in the config file and a `static Set<String>` in this class which references them, and finally a Set#contains here.

Also fine with you placing a `String KEY_PREFIX_LOOT_RECOLOR = "sarcophagusRecolour"` and using `keyName = KEY_PREFIX_LOOT_RECOLOR + "White"`, then checking for startsWith that prefix as well.

At least some sort of reference-based comparison here in case the config key names change.